actioncable/app/assets/javascripts/action_cable/connection.coffee

```diff
@@ -90,18 +103,18 @@ class ActionCable.Connection
     open: ->
       ActionCable.log("WebSocket onopen event")
       @disconnected = false
-      @subscriptions.reload()
+      if not @isProtocolSupported()
+        ActionCable.log("WebSocket got '#{@getProtocol()}' as the chosen protocol. Stopping monitor and disconnecting.")
+        @close(allowReconnect: false)
+        return
+      ActionCable.log("WebSocket using '#{@getProtocol()}' subprotocol.")
 
-    close: ->
+    close: (event) ->
       ActionCable.log("WebSocket onclose event")
-      @disconnect()
+      return if @disconnected
+      @disconnected = true
```

Comment:
I think that should stay for now. It still seems to prevent some conditions where `onClose` is called by the websocket even though the websocket is already disconnected, which I think would be confusing to the developer.